### ISSUE TITLE
completed-docs checks too many types of variables

### ISSUE THREAD

**Issue by reduckted** [palantir/tslint#2912]
### Bug Report

- __TSLint version__: 5.4.3
- __TypeScript version__: 2.3.4
- __Running TSLint via__: Anything

#### TypeScript code being linted

```ts
const bar = 2;
try {
} catch (ex) {
}
for (let i = 0; i < 1; i++) { }
for (let i of []) { }
for (let p in {}) { }
function foo() {
    let x = 1;
}
```

with `tslint.json` configuration:

```json
{
    "rules": {
        "completed-docs": [true, "variables"]
    }
}
```

#### Actual behavior

Errors are produced for every variable:

    ERROR: index.ts[1, 7]: Documentation must exist for variables.
    ERROR: index.ts[3, 10]: Documentation must exist for variables.
    ERROR: index.ts[5, 10]: Documentation must exist for variables.
    ERROR: index.ts[6, 10]: Documentation must exist for variables.
    ERROR: index.ts[7, 10]: Documentation must exist for variables.
    ERROR: index.ts[9, 11]: Documentation must exist for variables.

#### Expected behavior

Only an error for the first variable that is declared at the file-level is produced:

    ERROR: index.ts[1, 7]: Documentation must exist for variables.

---

Variables in catch blocks and for loops can't have documentation, so they should never be checked. 

The variable inside the function is an interesting case. If you add documentation to it, the error goes away, but does anyone document there local variables like that? It seems like a strange thing to do.

In my opinion, the intent behind wanting variables to be documented, is because they are exported or "global" to the file they are in, and those types of variables would only be declared at the file-level.

If we're happy to restrict this rule to only check variables at the file-level, then I've got a PR that's ready to go.
